src/handlers/net/v2/NetStreamHandler.h

```diff
@@ -0,0 +1,252 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#pragma once
+
+#include "AbstractMetricsManager.h"
+#include "Corrade/Utility/Debug.h"
+#include "DnstapInputStream.h"
+#include "GeoDB.h"
+#include "MockInputStream.h"
+#include "PcapInputStream.h"
+#include "StreamHandler.h"
+#include <string>
+
+namespace visor::handler::net::v2 {
+
+using namespace visor::input::pcap;
+using namespace visor::input::dnstap;
+using namespace visor::input::mock;
+
+static constexpr const char *NET_SCHEMA{"net"};
+
+namespace group {
+enum NetMetrics : visor::MetricGroupIntType {
+    Counters,
+    Cardinality,
+    TopGeo,
+    TopIps
+};
+}
+
+struct NetworkPacket {
+    PacketDirection dir;
+    pcpp::ProtocolType l3;
+    pcpp::ProtocolType l4;
+    size_t payload_size;
+    bool syn_flag;
+    pcpp::IPv4Address ipv4_in;
+    pcpp::IPv4Address ipv4_out;
+    pcpp::IPv6Address ipv6_in;
+    pcpp::IPv6Address ipv6_out;
+
+    NetworkPacket(PacketDirection dir, pcpp::ProtocolType l3, pcpp::ProtocolType l4, size_t payload_size, bool syn_flag)
+        : dir(dir)
+        , l3(l3)
+        , l4(l4)
+        , payload_size(payload_size)
+        , syn_flag(syn_flag)
+    {
+    }
+};
+
+class NetworkMetricsBucket final : public visor::AbstractMetricsBucket
+{
+
+protected:
+    mutable std::shared_mutex _mutex;
+
+    Cardinality _srcIPCard;
+    Cardinality _dstIPCard;
+
+    TopN<visor::geo::City> _topGeoLoc;
+    TopN<std::string> _topASN;
+    TopN<uint32_t> _topIPv4;
+    TopN<std::string> _topIPv6;
+
+    // total numPackets is tracked in base class num_events
+    struct counters {
+        Counter UDP;
+        Counter TCP;
+        Counter OtherL4;
+        Counter IPv4;
+        Counter IPv6;
+        Counter TCP_SYN;
+        Counter total_in;
+        Counter total_out;
+        Counter total_unk;
+        Counter total;
+        Counter filtered;
+        counters()
+            : UDP(NET_SCHEMA, {"udp_packets"}, "Count of UDP packets")
+            , TCP(NET_SCHEMA, {"tcp_packets"}, "Count of TCP packets")
+            , OtherL4(NET_SCHEMA, {"other_l4_packets"}, "Count of packets which are not UDP or TCP")
+            , IPv4(NET_SCHEMA, {"ipv4_packets"}, "Count of IPv4 packets")
+            , IPv6(NET_SCHEMA, {"ipv6_packets"}, "Count of IPv6 packets")
+            , TCP_SYN(NET_SCHEMA, {"protocol", "tcp", "syn_packets"}, "Count of TCP SYN packets")
```

Comment:
how should we proper rename this metrics @rboucher-me  the current prometheus name became
`net_protocol_tcp_syn_packets`

Reply:
Ideally we would just name it `net_tcp_syn_packets`. Does the `protocol` have to be there?